src/settings/convertSettings.ts

```diff
@@ -0,0 +1,48 @@
+import { ConversionError } from "../errors/conversionError";
+import { ErrorSummary } from "../errors/errorSummary";
+import { EditorSettingConverter } from "./converter";
+import { convertSetting } from "./convertSetting";
+import { EditorSetting } from "./types";
+
+export type ConvertSettingsDependencies = {
+    converters: Map<string, EditorSettingConverter>;
+};
+
+export type SettingConversionResults = {
+    converted: Map<string, EditorSetting>;
+    failed: ErrorSummary[];
+    missing: Pick<EditorSetting, "settingName">[];
+};
+
+export type EditorConfigurationSettings = Record<string, any>;
+
+export const convertSettings = (
+    dependencies: ConvertSettingsDependencies,
+    rawEditorSettings: EditorConfigurationSettings,
+): SettingConversionResults => {
+    const converted = new Map<string, EditorSetting>();
+    const failed: ConversionError[] = [];
+    const missing: Pick<EditorSetting, "settingName">[] = [];
+
+    for (const [settingName, value] of Object.entries(rawEditorSettings)) {
+        const editorSetting = { settingName, value };
+        const conversion = convertSetting(editorSetting, dependencies.converters);
+
+        if (conversion === undefined) {
+            const { settingName } = editorSetting;
+            missing.push({ settingName });
```

Comment:
Wait, I don't think we want to be logging all these ones as `missing`. A log file from TSLint's `.vscode/settings.json` gives these settings complaints:

```
editor.insertSpaces does not yet have an ESLint equivalent.
editor.tabSize does not yet have an ESLint equivalent.
editor.wordWrap does not yet have an ESLint equivalent.
editor.wordWrapColumn does not yet have an ESLint equivalent.
editor.wrappingIndent does not yet have an ESLint equivalent.
files.eol does not yet have an ESLint equivalent.
search.exclude does not yet have an ESLint equivalent.
typescript.tsdk does not yet have an ESLint equivalent.
```

It's really only the editor settings in https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin#configuration that need to be mentioned, right?

Reply:
That's true. Previously tried to convert the entire **editor configuration** (`.vscode/settings.json`). Changed it to handle only the **editor settings** from the **editor configuration** (those starting with `editor.`).

This change also indicates the naming as mentioned below: The **editor configuration** is the entire file like `.vscode/settings` while the **editor setting** are only those keys in the config that start with `editor.`.